### PR TITLE
Bicep enhancements

### LINKED ISSUE #13: DBHostName Web App setting is incorrect
It refers to the name of the MySQL flexible server resource, not to the server's FQDN

### LINKED ISSUE #20: Configure App Service VNet integration

### PR DESCRIPTION
**Pull request by SvenAelterman**

### CODE REVIEW

**Review by matheuscarboni — APPROVED** [review]

File: deploy.ps1
Comment (sjyang18) [inline deploy.ps1:1]: I know this is merged. But, wonder if we may change the name of this file. We have another file called deploy.sh, and at first,  I thought they are doing the same deployment for different environments (one for Windows, one for Linux).

File: deploy.ps1 [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+#Requires -Modules "Az"
+#Requires -PSEdition Core
+
+# Use these parameters to customize the deployment instead of modifying the default parameter values
+[CmdletBinding()]
+Param(
+    [ValidateSet('eastus2', 'eastus')]
+    [Parameter()]
+    [string]$Location = 'eastus',
```
Comment (sjyang18) [inline deploy.ps1:11]: If we are thinking of sharing this script to customers, how about we change this a required parameter (i.e., no default value) and remove ValidateSet? Or we may get the input from user with an expected values like?

```
$location = Read-Host -Prompt "Provide the location (like, eastus, westus, eastus2, westus2,etc)?"
```

File: modules/kv/kv.bicep [hunk 2]
```diff
@@ -64,6 +64,7 @@ resource keyVaultSecrets 'Microsoft.KeyVault/vaults/secrets@2021-11-01-preview'
 
 resource roleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = [for roleAssignment in roleAssignments: {
   scope: keyVault
+  // TODO: Must include the object ID otherwise conflicts occur
```
Comment (sjyang18) [inline modules/kv/kv.bicep:67]: To avoid the conflict, we may need to include resourceID + location as well.

File: azDeploySecureSub.bicep [hunk 3]
```diff
@@ -1,32 +1,44 @@
 targetScope = 'subscription'
 
+@description('The Azure region to target for the deployment. Replaces {loc} in namingConvention.')
 @allowed([
   'eastus'
   'westus'
 ])
 param location string = 'eastus'
```
Comment (sjyang18) [inline azDeploySecureSub.bicep:8]: required parameter? How about we remove the default value.

**Pull request merged**